tests/shoppingSite.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { test, expect } from '@playwright/test'
+import { ShoppingSitePage } from '../Pages/MainPage'
+
+test.describe('Ecommerce Site', () => {
+  let shoppingSite: ShoppingSitePage
+
+  test.beforeEach(async ({ page }) => {
+    shoppingSite = new ShoppingSitePage(page)
+    await shoppingSite.navigateToSite()
+  })
+
+  test('Verify stock levels are displayed correctly for each size filter', async ({
+    page,
+  }) => {
+    const expectedStockLevels = {
+      XS: 1,
+      S: 2,
+      M: 1,
+      ML: 2,
+      L: 10,
+      XL: 10,
+      XXL: 4,
+    }
+    for (const [size, expectedStock] of Object.entries(expectedStockLevels)) {
+      await test.step(`Given the customer has navigated to the site`, async () => {})
+      await test.step('When the customer selects a size filter to view available shirt models.', async () => {
+        await shoppingSite.selectSize(size)
+      })
+      await test.step(`Then the stock level for the "${size}" size should be ${expectedStock}.`, async () => {
+        const actualStock = await shoppingSite.getStockLevel()
+        console.log(`The stock level of ${size} is ${actualStock}.`)
+        expect(actualStock).toBe(expectedStock)
+      })
+    }
+  })
+
+  test('Verify adding items to the cart works', async ({ page }) => {
+    await test.step('Given the customer has selected 6 items to add to the cart', async () => {
```

Comment:
We can skip an empty step by stating in the next one that the customer is performing the action from the page he is supposed to be at.